# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: 726458f976ff
# Full Hash: 726458f976fff2b05bb0eaf7e940cd7be06d75f5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 1: Use BigBuffer for IPCDataTransfer, r=edgar
#   
#   The IPCDataTransfer type is used to transfer Clipboard/Drag & Drop payloads
#   over IPC to allow them to be written to or read from the relevant system
#   interfaces. Previously, the system which was used was somewhat complex, and
# ==============================================================================

diff -r 1aa5ec22f47a -r 726458f976ff dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Tue Aug 02 17:08:59 2022 +0000
+++ b/dom/ipc/BrowserParent.h	Tue Aug 02 17:15:41 2022 +0000
@@ -386,7 +386,7 @@
 
   mozilla::ipc::IPCResult RecvSetCursor(
       const nsCursor& aValue, const bool& aHasCustomCursor,
-      const nsCString& aCursorData, const uint32_t& aWidth,
+      Maybe<BigBuffer>&& aCursorData, const uint32_t& aWidth,
       const uint32_t& aHeight, const float& aResolutionX,
       const float& aResolutionY, const uint32_t& aStride,
       const gfx::SurfaceFormat& aFormat, const uint32_t& aHotspotX,
@@ -614,7 +614,7 @@
 
   bool SendInsertText(const nsString& aStringToInsert);
 
-  bool SendPasteTransferable(const IPCDataTransfer& aDataTransfer,
+  bool SendPasteTransferable(IPCDataTransfer&& aDataTransfer,
                              const bool& aIsPrivateData,
                              nsIPrincipal* aRequestingPrincipal,
                              const nsContentPolicyType& aContentPolicyType);